api_end_end_testing/src/test/java/tds/user/api/tests/CreateUserTest.java

```diff
@@ -0,0 +1,247 @@
+package tds.user.api.tests;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.jayway.restassured.RestAssured.*;
+
+import com.jayway.restassured.RestAssured;
+import com.jayway.restassured.response.Header;
+import com.jayway.restassured.http.ContentType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.testng.annotations.Test;
+import tds.user.api.model.UserInfo;
+import tds.user.api.model.RoleAssociation;
+
+/*
+    Authenticate user
+    Testing HTTP POST of https://sso-deployment.sbtds.org/auth/oauth2/access_token?realm=/sbac
+    and success status 200
+ */
+public class CreateUserTest extends BaseUri{
+    String accessToken = null;
+    String email = null;
+
+    @Test
+    public void getAuthenticationCode() {
+        RestAssured.baseURI = authenticateURI;
+
+        Header header = new Header("Content-Type", "application/x-www-form-urlencoded");
+
+        accessToken = given()
+            .contentType("application/x-www-form-urlencoded")
+            .header(header)
+            .queryParam("realm", "/sbac")
+            .formParam("client_id", "pm")
+            .formParam("client_secret", "sbac12345")
+            .formParam("grant_type", "password")
+            .formParam("password", "password")
+            .formParam("username", "prime.user@example.com")
+            .when()
+            .post("/auth/oauth2/access_token")
+            .then()
+            .statusCode(200)
+            .extract()
+            .path("access_token");
+    }
+
+    /*
+        Create user
+        Testing HTTP POST of /rest/external/user, 201 success item created
+    */
+    @Test(dependsOnMethods = "getAuthenticationCode")
+    public void createUser() {
+        email = "betsy.ross@example.com";
+
+        RestAssured.baseURI = userUri;
+
+        List<RoleAssociation> roleAssociations = new ArrayList<RoleAssociation>();
+        roleAssociations.add(new RoleAssociation("Administrator", "CLIENT", "98765"));
+        roleAssociations.add(new RoleAssociation("Administrator", "CLIENT", "98765"));
+
+        Header authHeader = new Header("Authorization", "Bearer " + accessToken);
+
+        UserInfo userInfo = new UserInfo(email, "Betsy", "Ross", "800-332-4747", roleAssociations);
+
+        ObjectMapper mapper = new ObjectMapper();
+        String jsonInString = null;
```

Comment:
Lines 67-74 seem to be debug code or left over code from an earlier strategy to test the endpoints.  I suggest that you can remove them because jsonInString doesn't seem to be used anymore in the code starting at line 76.